src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Context/FormattingContext.cs

```diff
@@ -41,7 +41,7 @@ internal partial class FormattingContext
         // anchor token to anchor data map.
         // unlike anchorTree that would return anchor data for given span in the tree, it will return
         // anchorData based on key which is anchor token.
-        private readonly Dictionary<SyntaxToken, AnchorData> _anchorBaseTokenMap;
+        private readonly SegmentedDictionary<SyntaxToken, AnchorData> _anchorBaseTokenMap;
```

Comment:
📝 This was showing a few LOH allocations before dotnet/roslyn#42863, but many more after it. Fixing it was super easy.